Controller/Component/Auth/ApiAuthenticate.php

```diff
@@ -91,4 +91,16 @@ protected function _getAuthString(CakeRequest $request) {
 
 		return str_ireplace('Bearer ', '', $headers[$this->_authHeaderName]);
 	}
+
+	private function _getallheaders() {
+		foreach($_SERVER as $key=>$value) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Do not access "$_SERVER" directly. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS2043)